ingest/src/main/scala/hydra.ingest/http/TopicDeletionEndpoint.scala

```diff
@@ -0,0 +1,146 @@
+package hydra.ingest.http
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.directives.Credentials
+import akka.http.scaladsl.server.{ExceptionHandler, Route}
+import cats.data.{NonEmptyList, Validated, ValidatedNel}
+import hydra.ingest.programs.{DeleteTopicError, KafkaDeletionErrors, SchemaDeletionErrors, TopicDeletionProgram}
+import hydra.common.util.Futurable
+import hydra.core.http.RouteSupport
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+import hydra.core.monitor.HydraMetrics.addPromHttpMetric
+import hydra.ingest.programs.TopicDeletionProgram.SchemaDeleteTopicErrorList
+
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+final class TopicDeletionEndpoint[F[_]: Futurable] (deletionProgram: TopicDeletionProgram[F], deletionPassword: String)
+  extends RouteSupport with
+    DefaultJsonProtocol with
+    SprayJsonSupport {
+
+  implicit val endpointFormat = jsonFormat2(DeletionEndpointResponse.apply)
+  case class DeletionEndpointResponse(topicOrSubject: String, message: String)
```

Comment:
These should be private and put into a companion object rather than direct in the class here. I am guessing the format can also be set as private

Reply:
Oh and also, final on the case class since extending case classes is weird.